inventory-repository/src/main/scala/com/normation/inventory/ldap/core/InventoryMapper.scala

```diff
@@ -697,35 +725,39 @@ class InventoryMapper(
     root +=! (A_POLICY_SERVER_UUID, server.main.policyServerId.value)
-    root.setOpt(server.ram, A_OS_RAM, { m: MemorySize => m.size.toString })
-    root.setOpt(server.swap, A_OS_SWAP, { m: MemorySize => m.size.toString })
-    root.setOpt(server.archDescription, A_ARCH, { x: String => x })
-    root.setOpt(server.lastLoggedUser, A_LAST_LOGGED_USER, { x: String => x })
+    root.setOpt(server.ram               , A_OS_RAM, { m: MemorySize => m.size.toString })
+    root.setOpt(server.swap              , A_OS_SWAP, { m: MemorySize => m.size.toString })
+    root.setOpt(server.archDescription   , A_ARCH, { x: String => x })
+    root.setOpt(server.lastLoggedUser    , A_LAST_LOGGED_USER, { x: String => x })
     root.setOpt(server.lastLoggedUserTime, A_LAST_LOGGED_USER_TIME, { x: DateTime => GeneralizedTime(x).toString })
-    root.setOpt(server.inventoryDate, A_INVENTORY_DATE, { x: DateTime => GeneralizedTime(x).toString })
-    root.setOpt(server.receiveDate, A_RECEIVE_DATE, { x: DateTime => GeneralizedTime(x).toString })
-    root +=! (A_AGENTS_NAME, server.agents.map(x => x.toJsonString):_*)
-    root +=! (A_SOFTWARE_DN, server.softwareIds.map(x => dit.SOFTWARE.SOFT.dn(x).toString):_*)
-    root +=! (A_EV, server.environmentVariables.map(x => Serialization.write(x)):_*)
-    root +=! (A_PROCESS, server.processes.map(x => Serialization.write(x)):_*)
-    root +=! (A_LIST_OF_IP, server.serverIps.distinct:_*)
+    root.setOpt(server.inventoryDate     , A_INVENTORY_DATE, { x: DateTime => GeneralizedTime(x).toString })
+    root.setOpt(server.receiveDate       , A_RECEIVE_DATE, { x: DateTime => GeneralizedTime(x).toString })
+    root +=! (A_AGENTS_NAME       , server.agents.map(x => x.toJsonString):_*)
+    root +=! (A_SOFTWARE_DN       , server.softwareIds.map(x => dit.SOFTWARE.SOFT.dn(x).toString):_*)
+    root +=! (A_EV                , server.environmentVariables.map(x => Serialization.write(x)):_*)
+    root +=! (A_PROCESS           , server.processes.map(x => Serialization.write(x)):_*)
+    root +=! (A_LIST_OF_IP        , server.serverIps.distinct:_*)
     //we don't know their dit...
-    root +=! (A_CONTAINER_DN, server.machineId.map { case (id, status) =>
+    root +=! (A_CONTAINER_DN      , server.machineId.map { case (id, status) =>
       ditService.getDit(status).MACHINES.MACHINE.dn(id).toString
     }.toSeq:_*)
-    root +=! (A_ACCOUNT, server.accounts:_*)
-    root +=! (A_SERVER_ROLE, server.serverRoles.toSeq.map(_.value):_*)
+    root +=! (A_ACCOUNT           , server.accounts:_*)
+    root +=! (A_SERVER_ROLE       , server.serverRoles.toSeq.map(_.value):_*)
     server.timezone.foreach { timezone =>
-      root +=! (A_TIMEZONE_NAME, timezone.name)
-      root +=! (A_TIMEZONE_OFFSET, timezone.offset)
+      root +=! (A_TIMEZONE_NAME   , timezone.name)
+      root +=! (A_TIMEZONE_OFFSET , timezone.offset)
+    }
+    server.customProperties.foreach { cp =>
+      import CustomPropertiesSerialization.Serialise
```

Comment:
just a remark: is there a penalty to have imports within a foreach, or is it managed at compile time ?

Reply:
compile time. It might help compile time with scoping of import, but more for things with a lot of implicits.